File: src/test/java/nextstep/ladder/domain/laddergame/position/PlayerPositionsTest.java
package nextstep.ladder.domain.laddergame.position;

import nextstep.ladder.domain.ladder.Ladder;
import nextstep.ladder.domain.ladder.LadderTest;
import nextstep.ladder.domain.player.Players;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

class PlayerPositionsTest {

    private PlayerPositions playerPositions;

    @BeforeEach
    void setUpPlayerPositions() {
        Ladder ladder = LadderTest.getLadder();

        List<Positions> positionsList = new ArrayList<>();

        for (int i = 0; i < ladder.getPlayers().size(); i++) {
            Positions positions = new Positions();
            positionsList.add(positions.add(ladder.getLines(), i));
        }

        playerPositions = new PlayerPositions(new Players(ladder.getPlayers()), positionsList);
    }

    @Test
    @DisplayName("플레이어의 마지막 위치를 반환한다")
    void getLastPosition() {
        Assertions.assertThat(playerPositions.getLastPosition("test2")).isEqualTo(1);
        Assertions.assertThat(playerPositions.getLastPosition("test4")).isEqualTo(0);
    }

    @Test
    void getPlayers() {
        Assertions.assertThat(playerPositions.getPlayers()).isNotNull();
        Assertions.assertThat(playerPositions.getPlayers().size()).isEqualTo(4);
    }

    @Test
    void getPositions() {
        Assertions.assertThat(playerPositions.getPositions()).isNotNull();
        Assertions.assertThat(playerPositions.getPositions().size()).isEqualTo(4);
    }
}

File: src/test/kotlin/ladder/domain/PlayerPositionsTest.kt
package ladder.domain

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test

class PlayerPositionsTest {
    private lateinit var playerPositions: PlayerPositions

    @Before
    fun setup() {
        playerPositions = PlayerPositions(PLAYERS, LADDER)
    }

    @Test
    fun `세번째 포지션값 확인`() {
        assertThat(playerPositions.size()).isEqualTo(3)
    }

    @Test
    fun `플레이어 결과 확인`() {
        val dave = playerPositions.get(Player("Dave"))
        val lee = playerPositions.get(Player("Lee"))
        val kim = playerPositions.get(Player("Kim"))
        assertThat(dave).isEqualTo(2)
        assertThat(lee).isEqualTo(0)
        assertThat(kim).isEqualTo(1)
    }

    @Test(expected = IllegalArgumentException::class)
    fun `존재하지 않는 유저 입력시 에러`() {
        playerPositions.get(Player("김아무개"))
    }
}